src/parsers/pm_io/psse.jl

```diff
@@ -21,29 +21,33 @@ function _init_bus!(bus::Dict{String, Any}, id::Int)
     return
 end
 
+function _get_bus_value(bus_i::Int, field::String, pm_bus_data::Array)
+    for bus in pm_bus_data
+        if bus["index"] == bus_i
+            return bus[field]
+        end
+    end
+    @info("Could not find bus $bus_i, returning 0 for field $field")
+    return 0
+end
+
+function _get_bus_value(bus_i::Int, field::String, pm_bus_data::Dict{String, Any})
+    if !haskey(pm_bus_data["bus"], bus_i)
+        @info("Could not find bus $bus_i, returning 0 for field $field")
+        return 0
+    else
+        return pm_bus_data["bus"][bus_i][field]
+    end
+end
+
 """
     _get_bus_value(bus_i, field, pm_data)
 
 Returns the value of `field` of `bus_i` from the PowerModels data. Requires
 "bus" Dict to already be populated.
 """
-function _get_bus_value(bus_i, field, pm_data)
-    if isa(pm_data["bus"], Array)
-        for bus in pm_data["bus"]
-            if bus["index"] == bus_i
-                return bus[field]
-            end
-        end
-    elseif isa(pm_data["bus"], Dict)
-        for (k, bus) in pm_data["bus"]
-            if bus["index"] == bus_i
-                return bus[field]
-            end
-        end
-    end
-
-    @info("Could not find bus $bus_i, returning 0 for field $field")
-    return 0
+function _get_bus_value(bus_i::Int, field::String, pm_data)
+    return _get_bus_value(bus_i, field, pm_data["bus"])
```

Comment:
This only calls the Array version, right?

Reply:
this dispatches depending on wether `pm_data["bus"]` is an Array or a Dict.

Reply:
Perhaps you could make two different function names: one that takes the full pm_data and one that dispatches on the type of the bus container.

Reply:
Changes to `_find_bus_value`